src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -138,9 +139,37 @@ public static boolean onDrawBlockHighlight(WorldRenderer context, ActiveRenderIn
         return MinecraftForge.EVENT_BUS.post(new DrawHighlightEvent(context, info, target, partialTicks, matrix, buffers));
     }
 
-    public static void dispatchRenderLast(WorldRenderer context, MatrixStack mat, float partialTicks, Matrix4f projectionMatrix, long finishTimeNano)
+    private static float partialTicksHolder;
+    private static ClippingHelper clippinghelperHolder;
+    private static ActiveRenderInfo activeRenderInfoInHolder;
```

Comment:
This seems weird, do we really need this context on thw WorldBlockLayerEvent?

Reply:
ClippingHelper nad partialTicks is questionable, you definitely need ActiveRenderInfo for positioning - might make it event special and only pass in d0-d2 from vanilla (since that's usually the only thing you need)